### ISSUE TITLE
pkcon fails when repo return 404

### ISSUE THREAD

**Issue by razzeee** [PackageKit/PackageKit#814]
I originally raised this on g-s https://gitlab.gnome.org/GNOME/gnome-software/-/issues/2737

But it seems to be a packagekit problem

```
❯ pkcon update
Getting updates                         [=========================]         
Finished                                [                         ] (0%)  Command failed: cannot update repo 'insync': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried; Last error: Status code: 404 for http://yum.insync.io/fedora/41/repodata/repomd.xml (IP: 138.199.37.227)


```

**Comment by sidt4** [comment]
The error message [is from `libdnf`](https://github.com/rpm-software-management/libdnf/blob/dnf-4-master/libdnf/dnf-repo.cpp#L1900-L1905).

For anyone interested, here is the control flow:

- https://github.com/PackageKit/PackageKit/blob/main/backends/dnf/pk-backend-dnf.c#L1672C17-L1715C3
   - https://github.com/PackageKit/PackageKit/blob/main/backends/dnf/pk-backend-dnf-refresh.c#L92-L95
      - https://github.com/rpm-software-management/libdnf/blob/dnf-4-master/libdnf/dnf-repo.cpp#L1900-L1905

**Comment by razzeee** [comment]
I guess that means, I should raise the issue there?

**Comment by sidt4** [comment]
No. The issue here is not the error from `libdnf`, but how it's handled in PackageKit.

So, I think it's still a PackageKit issue. Only thing we need to figure is, if it's specific to `DNF` (Fedora) plugin, or it happens with other backends APT (Debian), ALPM (Arch) etc too.
